### ISSUE TITLE
Watch app fails to open or crashes when trying to update the graph

### ISSUE THREAD

**Issue by threegirldad** [nightscout/nightguard#217]
Installed nightguard yesterday (30 Jan 2023). The watch app worked perfectly for about 20 hours, then stopped working. As the title says, tapping the complication either results in nothing happening at all, or the app opening briefly, and then crashing when it tries to update the graph.  Meanwhile, the phone app is still working just fine.  The complication on the watch face now looks like this:

??:??
[---]

Ignore the "[" and "]" characters.  I added those so that the editor won't convert the dashes to a solid line.

Apple Watch Series 8
WatchOS 9.3 (20S648)

**Comment by dhermanns** [comment]
Hi! Does the problem persist after deinstalling/reinstalling the App? If yes, you could leave me a logon token. I should be able to reproduce it on my site then, too.

**Comment by threegirldad** [comment]
Thanks for asking.  I went ahead and uninstalled/last night last night.  So far, so good.  Curious to see if anything happens again when I get closer to running the app for 24 hours.

**Comment by threegirldad** [comment]
The issue has now occurred a second time, about 12 hours after uninstalling/reinstalling.

Logon token:


https://a877.ns.gluroo.com:39893<https://a877.ns.gluroo.com:39893/>/?token=a877173d-20b1-4fc3-9e66-327041341915

________________________________

**Comment by threegirldad** [comment]
Also, unlike the first uninstall/reinstall, the watch complication is no longer working after this uninstall/reinstall.  The app opens, but does not display any data.

I had to uninstall the app from the watch, then uninstall/reinstall on the phone a third time, and now the watch app is working again for the moment.

**Comment by dhermanns** [comment]
Your link takes me to a gluroo welcome page.
Only nightscout backends are supported.

**Comment by threegirldad** [comment]
For clarification:

https://gluroo.com/blog/glucrew/nightscout-heroku-alternative-free/

If I understand you last comment correctly, I need to switch to a "conventional" Nightscout server setup?

Thanks for your time.

**Comment by dhermanns** [comment]
I would bet that gluroo is not 100% Api compatible or at least sending unexpected data. Looks like that is causing trouble on the watch part.

So you should try to reproduce this with a standard nightscout backend.

**Comment by threegirldad** [comment]
Thanks for explaining.  And my apologies for taking up your time.

I will look into switching to a standard Nightscout backend.

**Comment by dhermanns** [comment]
No problem. Had a look and looks like you get blood glucose values every minute? Maybe that is causing trouble, since most people get values every 5 minutes.

**Comment by threegirldad** [comment]
Thanks for your question.  I wondered if that might be part of the problem. I use the Libre 3, which records a reading every minute, not every five minutes.

**Comment by dhermanns** [comment]
Checked in simulator and it is working fine so far.

**Comment by threegirldad** [comment]
Thank you again for your time.  I appreciate it very much.

If this problem happens again, I will try changing the upload interval to five minutes instead of one, and if there is still an issue, I will try to move to a conventional Nightscout backend.

**Comment by threegirldad** [comment]
Hello again.  I am now using one of the Nightscout hosting services mentioned on The Nightscout Project's site.  The service is 
 Nightscout.pro.

https://nightscout.pro/

I asked for details about their implementation, and received this reply:

"The implementation of Nightscout we use is pulled from the docker image from the official cgm-remote-monitor Github repo, and we have made absolutely no changes to it."

I have been running Nightguard for several days now using Nightscout.pro as a source.  Here are the results so far:

--The watch app has never crashed.  However, unlike when I was using Gluroo, Nightguard is now only updating every five minutes instead of every minute.  So, at the moment, I have no way of knowing if the watch app was crashing due to non-compliant data in Gluroo, or due to being overwhelmed by a reading every minute instead of every five minutes.

--When I used Gluroo as my source, Nightguard displayed a rolling 9-hour graph.  Using Nightscout.pro as a source, Nightguard is displaying a rolling 3-hour graph.

Here is my Nightscout.pro URL: https://superdave.nightscout.pro/?token=a877173d-20b1-4fc3-9e66-327041341915

Thanks again for your time.
